feat: Allow markdown title and description. (#12697)
argoproj/argo-workflows: docs/title-and-description.md

```

The above manifest will render as a row like the below image:

Workflows
![Title and Description Example](assets/workflow-title-and-description.png)

Cron Workflows
![Title and Description Example](assets/cron-workflow-title-and-description.png)

Workflow Templates
![Title and Description Example](assets/workflow-template-title-and-description.png)

Cluster Workflow Templates

> ![Title and Description Example](assets/cluster-workflow-template-title-and-description.png)

Reviewer: I think you misunderstood the direction I gave in my [previous comment](https://github.com/argoproj/argo-workflows/pull/12697#pullrequestreview-1899688100) -- I said to put these in a _new section_.

It's (significantly) easier to understand a single example than 4 in a row like this.
The new section also needs a version specifier (`> v3.6 and after`). You currently put these 3 screenshots in a section that says `> v3.4.4 and after`, which is not correct for these 3.

One screenshot for each would also suffice; no need to repeat the non-markdown, markdown, and complex, as those have already been described. Can have just 1 markdown example for each -- let's use the most complex one, the one with "Test Title".
Should also provide a small YAML example for each like the simple markdown one.

Author: Thanks for the review and the detailed explanation.
I have made the modifications, could you please check?

```

The above examples will render as rows like the below image:
![More Markdown Examples](assets/title-and-description-markdown-complex-workflow.png)

> ### Cluster Workflow Template

Reviewer: ```suggestion
### For `ClusterWorkflowTemplates`

> v3.6 and after

You can also add the `workflows.argoproj.io/title` and `workflows.argoproj.io/description` annotations with embedded markdown to a `ClusterWorkflowTemplate` to display in the list:
```

- Add the word "For" as a clarifier
  - In particular, headings pop up in docs searches, so plain "ClusterWorkflowTemplate" could be confusing.
  - I feel like there could be a better heading, but I'm not thinking of one off the top of my head right now
- You can use code backticks for resource names, which will cause it to be ignored by the spell checker.
  - It also is one word, so if you din't use backticks, you should add it to the [`.spelling`](https://github.com/argoproj/argo-workflows/blob/7286d499a60e38e226365ac0300571963379e00c/.spelling) file
    - (Note that CronWorkflow and WorkflowTemplate are already in that file, just not ClusterWorkflowTemplate)
- Add the version specifier, as I've mentioned before
- Introduce/describe the example, similar to the other existing sections

Reviewer: Also can you move this ClusterWorkflowTemplate section to the bottom? It is the least used and generally should be after WorkflowTemplates as well

```yaml
apiVersion: argoproj.io/v1alpha1
kind: ClusterWorkflowTemplate
metadata:
  name: my-cluster-workflow-template
  annotations:
    workflows.argoproj.io/title: '**Test Title**'
    workflows.argoproj.io/description: |
      `This is a simple hello world example.`
      You can also run it in Python: https://couler-proj.github.io/couler/examples/#hello-world
```

> The above examples will render as rows like the below image:

Reviewer: ```suggestion
The above manifest will render as a row like the below image:
```

singular "example" and singular "row". the existing doc uses "manifest" when there's a `kind` specified, so also use the term "manifest" here as well

> ![More Markdown Examples](assets/title-and-description-markdown-cluster-workflow-template.png)

Reviewer: The `alt` text should be different for each of these (and, in general, for every image on a page)
```suggestion
![ClusterWorkflowTemplate Example](assets/title-and-description-markdown-cluster-workflow-template.png)
```
also there's only a single example here, so it should be singular "example", not plural "examples"

> ### Cron Workflow

Reviewer: ```suggestion
### For `CronWorkflows`

> v3.6 and after

You can also add the `workflows.argoproj.io/title` and `workflows.argoproj.io/description` annotations with embedded markdown to a `CronWorkflow` to display in the list:
```
same as above

```yaml
apiVersion: argoproj.io/v1alpha1
kind: CronWorkflow
metadata:
  name: my-cron-workflow
  annotations:
    workflows.argoproj.io/title: '**Test Title**'
    workflows.argoproj.io/description: |
      `This is a simple hello world example.`
      You can also run it in Python: https://couler-proj.github.io/couler/examples/#hello-world
```

> The above examples will render as rows like the below image:

Reviewer: ```suggestion
The above manifest will render as a row like the below image:
```

same as above

> ![More Markdown Examples](assets/title-and-description-markdown-cron-workflow.png)

Reviewer: ```suggestion
![CronWorkflow Example](assets/title-and-description-markdown-cron-workflow.png)
```
same as above

> ### Workflow Template

Reviewer: ```suggestion
### For `WorkflowTemplates`

> v3.6 and after

You can also add the `workflows.argoproj.io/title` and `workflows.argoproj.io/description` annotations with embedded markdown to a `WorkflowTemplate` to display in the list:
```

same as above

```yaml
apiVersion: argoproj.io/v1alpha1
kind: WorkflowTemplate
metadata:
  name: my-workflow-template
  annotations:
    workflows.argoproj.io/title: '**Test Title**'
    workflows.argoproj.io/description: |
      `This is a simple hello world example.`
      You can also run it in Python: https://couler-proj.github.io/couler/examples/#hello-world
```

> The above examples will render as rows like the below image:

Reviewer: ```suggestion
The above manifest will render as a row like the below image:
```

same as above

> ![More Markdown Examples](assets/title-and-description-markdown-workflow-template.png)

Reviewer: ```suggestion
![WorkflowTemplate Example](assets/title-and-description-markdown-workflow-template.png)
```
same as above
